src/tests/suite/daos_obj_ec.c

```diff
@@ -285,16 +285,15 @@ ec_rec_list_punch(void **state)
 	daos_obj_id_t	oid;
 	int		num_rec;
 	int		i;
+	daos_recx_t	recx;
```

Comment:
NIT, not sure to understand why it has to be moved here

Reply:
IIRC, I moved this out of caution, because `recx` was referenced by `req` in a way similar to `data`.